qute_style/qute_style.py

```diff
@@ -390,9 +390,42 @@ def drawPrimitive(  # pylint: disable=invalid-name
                 option, (QStyleOptionButton, QStyleOptionViewItem)
             )
             self._draw_primitive_indicator_checkbox(option, painter)
+        elif element == self.PE_IndicatorBranch:
+            self._draw_branch(option, painter)
         else:  # pragma: no cover
             super().drawPrimitive(element, option, painter, widget)
 
+    @staticmethod
+    def _get_branch_color(option: QStyleOption) -> str:
+        """Select the right Branch color."""
+        if option.state & QStyle.State_MouseOver:
+            return get_color("context_hover")
+        return get_color("foreground")
+
+    @staticmethod
+    def _get_branch_icon(option: QStyleOption) -> str:
+        """Select the right Branch Icon."""
+        if option.state & QStyle.State_Open:
+            return ":/svg_icons/arrow_down.svg"
+        return ":/svg_icons/arrow_right.svg"
+
+    def _draw_branch(self, option: QStyleOption, painter: QPainter) -> None:
+        """Draw the branch arrow."""
+        if not option.state & QStyle.State_Children:
+            return
+
+        rect_width_height = int(
```

Comment:
There is no point of getting the average for the side of this rect, you simply could take the height value. To solve the issue with the space between icon and text that you still have one way is to adjust the rects x starting point.